### PR TITLE
replace utcnow with timezone naive utc call

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

There are two separate issues:

1. datetime.utcnow() is deprecated
2. we are using timezone naive time in lots of places

At first I tried to replace datetime.utcnow() with timezone-aware UTC time, but there were a hundred test errors and in fixing them, I found them getting more and more obscuring.  I think we have timezone-naive time in the db in spots, and probably also some questionable logic in the code.

You can see the issues if you go into app.utils and replace the call in utc_now() to use the timezone aware call instead of the naive one.

## Security Considerations

N/A

### CODE REVIEW

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM - We'll have to circle back and see about getting things updated to being timezone-aware - since the next thing Python is doing is removing the naive objects entirely. But this should technically avoid the deprecation warning ;)

File: app/utils.py [hunk 1]
```diff
@@ -132,3 +132,15 @@ def hilite(message):
     ansi_green = "\033[32m"
     ansi_reset = "\033[0m"
     return f"{ansi_green}{message}{ansi_reset}"
+
+
+def aware_utcnow():
+    return datetime.now(timezone.utc)
```
Comment (ccostino) [inline app/utils.py:138]: This method would be for the future and what `utc_now()` would start calling under the hood when getting back to being timezone aware?

File: notifications_utils/__init__.py [hunk 2]
```diff
@@ -23,3 +24,15 @@
 # and then later remove when performing tricky formatting operations
 MAGIC_SEQUENCE = "🇬🇧🐦✉️"
 magic_sequence_regex = re.compile(MAGIC_SEQUENCE)
+
+
+def aware_utcnow():
+    return datetime.now(timezone.utc)
```
Comment (ccostino) [inline notifications_utils/__init__.py:30]: Same question as above, but seeing this in the `notification_utils` code I see we're going to have our work cut out for us in terms of cleaning up duplication... 😆 😭

**Review by ccostino — COMMENTED** [review]
The changes here look fine to me, thanks @terrazoon!

However, to @xlorepdarkhelm's point, are we creating more work for ourselves by going down this route when Python itself is going to require timezone-aware in the future?

Perhaps not if the question I asked about the utility methods is true, as we'd only need to update and swap out in one or two spots.  If that's the case, then I'm fine moving forward with this, but I wanted to double check to be sure.  Thanks!

**Review by ccostino — APPROVED** [review]
Thank you, @terrazoon!

As noted in Slack, we'll come back to the timezone aware pieces in the future.

**Pull request merged**